proof-systems/src/darlin/tests/mod.rs

```diff
@@ -205,6 +232,27 @@ mod test {
 
         // Since the AHP failed, we are able to determine which proof verification has failed
         assert_eq!(result.unwrap_err().unwrap(), idx);
+
+        if fake_pcd.is_some() {
+
+            // Restore correct PCD
+            pcds[idx] = original_pcd;
+
+            let idx: usize = rng.gen_range(0, pcds.len());
+            pcds[idx] = fake_pcd.unwrap();
```

Comment:
Why we need to generate a new pcd with fake params and fake proof if we randomize the proof ? Can't we just randomize the proof ?

Reply:
When I am just randomizing the proof, it fails not in hard part